runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -152,14 +152,20 @@ inline void generateLoadJ9Class(TR::Node* node, TR::Register* j9class, TR::Regis
                    "Unexpected opCode for generateLoadJ9Class %s.", node->getOpCode().getName());
          break;
       }
+   if (node->getOpCode().isReadBar() || node->getOpCode().isWrtBar())
```

Comment:
It might be better to move this into `default` case of the switch statement, as checking whether the node is read/write barrier is unnecessary when its opcode falls into the other cases. In addition, the assertion can also be simplified.

Reply:
Changed according to comments.